Yv 801 governance implement earn rewards banner on tx page

AbdullayevAbdurashid opened the pull request:
https://emurgo.atlassian.net/browse/YV-801


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds an Earn Rewards banner to Tx History that guides users to delegate (stake pool or Yoroi DRep), fetching a top pool and building the corresponding transaction, with new i18n and config exports.
> 
> - **UI/UX**:
>   - **Earn Rewards Banner**: New `EarnRewardsBanner` component with CTA and dismiss in `.../EarnRewardsBanner/EarnRewardsBanner.tsx`.
>   - Integrated banner rendering into `TxHistory.tsx`.
> - **Logic/Flows**:
>   - `useEarnRewardsBanner`: Shows banner when user is on mainnet, not staked, not in governance, and a top pool is available; animates show/hide.
>   - `useEarnRewardsDelegation`: Builds delegation txs (stake pool delegation if `poolId` present, otherwise Yoroi DRep delegation, with optional stake key registration).
>   - `useTopStakePool`: Fetches biased top stake pool via `https://adapools.yoroiwallet.com/api/v1/pools`.
> - **Config/Exports**:
>   - Added `YOROI_TOP_STAKE_POOL_ID` and exported alongside `GOVERNANCE_YOROI_DREP_ID_HEX` in `@yoroi/staking/governance`.
> - **i18n**:
>   - Added EN strings and message definitions for the Earn Rewards banner; wired through `useStrings`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit efcdca72800a452a6a04c9337b3bade83d6c8b68. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

AbdullayevAbdurashid committed:
feature(mobile): add earn rewards banner and update governance configuration

- Introduced a new earn rewards banner in the transaction history screen.
- Added a test button for the earn rewards banner in development mode.
- Updated governance configuration to include the top Yoroi stake pool ID.
- Enhanced localization with new strings for the earn rewards banner.

AbdullayevAbdurashid committed:
refactor(mobile): update EarnRewardsBanner component and related hooks

- Replaced the illustration in the EarnRewardsBanner with a new governance image.
- Added a linear gradient background to enhance the visual appeal of the banner.
- Adjusted styles for better layout and spacing within the banner.
- Removed the test button for toggling the banner display in development mode.
- Simplified the useEarnRewardsBanner hook by removing unnecessary options and logic.
- Updated localization strings for the EarnRewardsBanner.

AbdullayevAbdurashid committed:
chore(mobile): remove unused GovernanceIllustration component

banklesss commented on mobile/src/features/Staking/Governance/useCases/EarnRewardsBanner/EarnRewardsBanner.tsx:
```diff
@@ -0,0 +1,160 @@
+  illustration: {
+    width: 100,
+    height: 120,
+  },
```
styles must follow repo conventions

banklesss commented on mobile/src/features/Staking/Governance/useCases/EarnRewardsBanner/useEarnRewardsBanner.tsx:
```diff
@@ -0,0 +1,108 @@
+    if (!wallet.isMainnet) return false
+    if (isParticipatingInStaking) return false
+    if (isParticipatingInGovernance) return false
+    if (isDismissed) return false
```
Unify this condition

banklesss commented on mobile/src/features/Staking/Governance/useCases/EarnRewardsBanner/useEarnRewardsBanner.tsx:
```diff
@@ -0,0 +1,108 @@
+
+  useWalletEvent(wallet, 'utxos', () => {
+    logger.info('Earn rewards banner: UTXOs changed, rechecking conditions')
+  })
```
unnecessary logger

banklesss commented on mobile/src/features/Staking/Governance/useCases/EarnRewardsBanner/useEarnRewardsBanner.tsx:
```diff
@@ -0,0 +1,108 @@
+    try {
+      logger.info(
+        'Earn rewards banner: Creating combined delegation transaction',
+      )
```
unnecessary logger

banklesss commented on mobile/src/features/Staking/Governance/useCases/EarnRewardsBanner/useEarnRewardsBanner.tsx:
```diff
@@ -0,0 +1,108 @@
+        context: 'delegate',
+        onSuccess: () => {
+          logger.info('Earn rewards banner: Transaction submitted successfully')
+        },
```
unnecessary logger

banklesss commented on mobile/src/features/Staking/Governance/useCases/EarnRewardsBanner/useEarnRewardsBanner.tsx:
```diff
@@ -0,0 +1,108 @@
+  ])
+
+  const handleDismiss = React.useCallback(() => {
+    logger.info('Earn rewards banner: Dismissed by user')
```
unnecessary logger

banklesss commented on mobile/src/features/Transactions/useCases/TxHistory/TxHistory.tsx:
```diff
@@ -99,6 +101,10 @@ export const TxHistory = () => {
 
+        {renderEarnRewardsBanner()}
+
+        {renderEarnRewardsBanner() != null && <Space.Height.md />}
```
simplify this. you are calling that function twice.

AbdullayevAbdurashid replied:
Updated

banklesss commented on mobile/packages/staking/governance/config.ts:
```diff
@@ -33,3 +33,8 @@ export const GOVERNANCE_ENDPOINTS: Readonly<
+
+// NOTE: Top Yoroi stake pool ID for mainnet
+// This should be configured based on the current top-performing Yoroi pool
+export const YOROI_TOP_STAKE_POOL_ID =
```
ask product about the needed pool

jorbuedo replied:
Check in the pool transition code, we have already a way there to get the pool we want users to delegate to.

AbdullayevAbdurashid replied:
Updated

banklesss replied:
so are these changes still needed @AbdullayevAbdurashid ?

banklesss commented on mobile/src/features/Staking/Governance/useCases/EarnRewardsBanner/EarnRewardsBanner.tsx:
```diff
@@ -0,0 +1,160 @@
+  )
+}
+
+const styles = StyleSheet.create({
```
styles must follow repo conventions

AbdullayevAbdurashid replied:
Updated

banklesss replied:
styles need to be inline. check other parts of the repo @AbdullayevAbdurashid

AbdullayevAbdurashid replied:
@banklesss  Done

AbdullayevAbdurashid committed:
fix(i18n): update EarnRewardsBanner title for clarity

AbdullayevAbdurashid committed:
feature(mobile): enhance EarnRewardsBanner with spacing and improve banner rendering in TxHistory

AbdullayevAbdurashid committed:
feature(mobile): integrate top stake pool logic into EarnRewardsBanner hook

AbdullayevAbdurashid committed:
feature(mobile): update EarnRewardsBanner to use dynamic gradient colors based on theme

- Added support for dark and light themes in the EarnRewardsBanner component.
- Updated the gradient colors to reflect the current theme, enhancing visual consistency.

banklesss commented:
@AbdullayevAbdurashid Take a look at the cursor recommendations

AbdullayevAbdurashid committed:
refactor(mobile): simplify EarnRewardsBanner styles and remove unused StyleSheet

jorbuedo commented:
I'll integrate your commits into another PR targeting mobile-experimental
